src/python/pants/option/custom_types.py

```diff
@@ -214,11 +223,13 @@ def create(cls, value) -> "ListValueComponent":
             appends = value
         elif value.startswith("[") or value.startswith("("):
             action = cls.REPLACE
-            appends = _convert(value, (list, tuple))
+            appends = _convert_list(value, member_type, is_enum)
```

Comment:
Btw, you can DRY this file through `functools.partial`:

```python
_convert_list_helper = partial(_convert_list, member_type=member_type, is_enum=is_enum)
appends = _convert_list_helper(value[1:])
```

Reply:
I think it makes for more confusing code... this code is complicated as it is, so using those kind of tricks just makes it harder to understand.